### ISSUE TITLE
zonedTimeToUtc not working

### ISSUE THREAD

**Issue by joshamb** [marnusw/date-fns-tz#145]
Im having some issues with **zonedTimeToUtc**, no matter what timezone I provide, it does nothing.

Here is an example:

```javascript
let timeiso = "2021-11-04T15:20:15Z"
let americaTimezone = "America/New_York" // the actual timezone of the above iso
let londonTimezone = "Europe/London" // for example

let aConv = zonedTimeToUtc(timeiso, americaTimezone); //Thu Nov 04 2021 10:20:15 GMT-0500 (America New_York Standard Time)
aConv.toISOString() //2021-11-04T10:20:15.000Z ||| EXPECTED WOUlD BE 5 HOURS LATER

let lConv = zonedTimeToUtc(timeiso, londonTimezone); // Thu Nov 04 2021 10:20:15 GMT-0500 (America New_York Standard Time)
lConv.toISOString() // 2021-11-04T10:20:15.000Z
```

In comparison to the docs, it shows **.toISOString()** converting the **zonedTimeToUtc** from 10am to 5pm, where in my case it stays the same:

```javascript
// DOCS:
const date = getDatePickerValue() // e.g. 2014-06-25 10:00:00 (picked in any time zone)
const timeZone = getTimeZoneValue() // e.g. America/Los_Angeles

const utcDate = zonedTimeToUtc(date, timeZone) // In June 10am in Los Angeles is 5pm UTC

postToServer(utcDate.toISOString(), timeZone) // post 2014-06-25T17:00:00.000Z, America/Los_Angeles
```

Any ideas what's going wrong?

Thanks in advance!

**Comment by Lzok** [comment]
@joshamb Have you got it working some way? I have the same trouble.

**Comment by marnusw** [comment]
The first problem is that `timeiso` is a UTC time, so if you want to format that for a specific time zone you have to use `utcToZonedTime` instead.

Once you've shifted to a different time zone you can't use `toISOString` anymore because the underlying UTC time would have been shifted to make it "look like" the date is in the time zone you specified. You should use the `format` function instead.

There have been reports of string parsing not always working correctly which I haven't investigated. If you still have issues please try converting `timeiso` to a `Date` first.

**Comment by marnusw** [comment]
Indeed the issue appears to be the string parsing done inside `zonedTimeToUTC`. The following works:

```js
let timeiso = '2021-12-04T15:20:15Z'
let americaTimezone = 'America/New_York' // the actual timezone of the above iso
let londonTimezone = 'Europe/London' // for example

let aConv = zonedTimeToUtc(new Date(timeiso), americaTimezone)

// On 4 November the USA is still on DST, so the offset is -04:00,
// therefore the time below has been advanced 4 hours
console.log(aConv.toISOString()) // 2021-11-04T19:20:15.000Z

let lConv = zonedTimeToUtc(new Date(timeiso), londonTimezone)

// On 4 November the UK is not on DST, so the local time and
// UTC time is the same
console.log(lConv.toISOString()) // 2021-11-04T15:20:15.000Z
```

**Comment by marnusw** [comment]
This was fixed in e7dc06945278 which will go out in `v1.2.0`.
